src/NuGet.Core/NuGet.Protocol/Converters/V3SearchResultsConverter.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using NuGet.Protocol.Model;
+
+namespace NuGet.Protocol.Converters
+{
+    internal class V3SearchResultsConverter : JsonConverter
+    {
+        private uint _take;
+
+        public V3SearchResultsConverter(uint take)
+        {
+            _take = take;
+        }
+
+        public override bool CanWrite => false;
+
+        public override bool CanConvert(Type objectType) => objectType == typeof(V3SearchResults);
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (objectType != typeof(V3SearchResults))
+            {
+                throw new InvalidOperationException();
+            }
+
+            if (reader.TokenType != JsonToken.StartObject)
+            {
+                throw new JsonException("Expected StartObject, found " + reader.TokenType);
+            }
+
+            var searchResults = new V3SearchResults();
+
+            var finished = false;
+
+            while (!finished)
+            {
+                reader.Read();
+
+                switch (reader.TokenType)
+                {
+                    case JsonToken.PropertyName:
+                        switch ((string)reader.Value)
+                        {
+                            case "totalHits":
+                                long totalHits = long.TryParse(reader.ReadAsString(), out totalHits) ? totalHits : 0;
+                                searchResults.TotalHits = totalHits;
+                                break;
+
+                            case "data":
+                                reader.Read();
+
+                                if (reader.TokenType != JsonToken.StartArray)
+                                {
+                                    throw new JsonException("data should be an array");
+                                }
+
+                                reader.Read();
+
+                                while (reader.TokenType != JsonToken.EndArray)
+                                {
+                                    var searchResult = JsonExtensions.JsonObjectSerializer.Deserialize<PackageSearchMetadata>(reader);
```

Comment:
Using a different serializer than the one passed in the method parameter goes against the normal converter conventions. I think it would be better for the caller to set up the serializer with all required converts, and then this converter doesn't need to use any external, static methods. (plus `JsonExtensions.JsonObjectSerializer` could be made private again)

Reply:
Refactored this code, so JsonExtensions.JsonObjectSerializer reverted back to private again.